File: app/src/main/java/wannabit/io/cosmostaion/dao/NameService.java
package wannabit.io.cosmostaion.dao;

import java.io.Serializable;

public class NameService implements Serializable {
    public NameServiceType type;
    public String name;
    public String address;

    public NameService(NameServiceType type, String name, String address) {
        this.type = type;
        this.name = name;
        this.address = address;
    }

    public enum NameServiceType{
        STARNAME, ICNS, STARGAZE, ICNS_STARGAZE;
    }
}

File: app/src/main/java/wannabit/io/cosmostaion/data/model/res/NameService.kt
package wannabit.io.cosmostaion.data.model.res

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.io.Serializable

@Parcelize
class NameService(
    var type: NameServiceType, var name: String, var address: String
) : Serializable, Parcelable {

    enum class NameServiceType {
        ICNS, STARGAZE, ARCHWAY, ENS
    }
}